pkg/api/v1alpha1/vspheredatacenterconfig_webhook.go

```diff
@@ -74,6 +75,15 @@ func (r *VSphereDatacenterConfig) ValidateUpdate(old runtime.Object) error {
 	if !ok {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected a VSphereDataCenterConfig but got a %T", old))
 	}
+	var allErrs field.ErrorList
+
+	if err := r.Validate(); err != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), r.Spec, err.Error()))
+	}
+
+	if len(allErrs) != 0 {
+		return apierrors.NewInvalid(GroupVersion.WithKind(VSphereDatacenterKind).GroupKind(), r.Name, allErrs)
+	}
```

Comment:
Because we have a pause annotation check after this, it returns from that condition. If we don't check for validate errors here and return, the results from Validate () won't get used.

Reply:
```suggestion
	if err := r.Validate(); err != nil {
		return apierrors.NewInvalid(
		    GroupVersion.WithKind(VSphereDatacenterKind).GroupKind(),
		    r.Name,
		    field.Invalid(field.NewPath("spec"), r.Spec, err.Error()),
	    )
	}
```